Workflow/Sources/WorkflowNode.swift

```diff
@@ -218,7 +218,14 @@ extension WorkflowNode {
         defer { observerCompletion?(state, output) }
 
         /// Apply the action to the current state
-        output = action.apply(toState: &state)
+        do {
+            // TODO: can we avoid instantiating a class here somehow?
```

Comment:
Is this TODO something you expect to address before merging?

Reply:
not really, more of a self-note. i'll update these to `FIXME` to make it more clear they are potential outstanding improvements, and not required pending tasks